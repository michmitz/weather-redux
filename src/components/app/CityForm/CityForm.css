form {
  display: inline-block;
}

.cityLabel {
  color: green;
  margin: 10px;
}

.submitButton {
  color: green;
}

textarea {
  width: 200px;
  height: 20px;
  padding: 2px;
  text-align: center;
  margin-left: 20px;
  margin-top: 20px;
}