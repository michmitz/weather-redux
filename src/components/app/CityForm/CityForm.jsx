import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchWeather, setSearch } from '../../../actions/weatherActions';

const CityForm = () => {
  const dispatch = useDispatch();
  const search = useSelector(state => state.search);
  
  const handleChange = ({ target }) => {
    dispatch(setSearch(target.value));
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    dispatch(fetchWeather(search));
  };

  return (
    <form onSubmit={handleSubmit}>
      <label>Enter a City and State:
        <textarea onChange={handleChange} placeholder="Portland, OR"></textarea>
      </label>
      <button>Submit</button>
    </form>
  );
};

export default CityForm;
