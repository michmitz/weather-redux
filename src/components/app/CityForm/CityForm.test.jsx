import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CityForm from './CityForm';
import { Provider } from 'react-redux';
import store from '../../../store';

describe('CityForm component', () => {
  afterEach(() => cleanup());
  it('renders CityForm', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <CityForm />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});