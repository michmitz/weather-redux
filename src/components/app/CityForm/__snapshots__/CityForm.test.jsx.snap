// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityForm component renders CityForm 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      id="heart"
    >
      <img
        src="/assets/rainbow.gif"
      />
    </div>
    <form>
      <input
        autocomplete="off"
        class="cityField"
        id="nme"
        name="name"
        required=""
        type="text"
      />
      <label
        for="nme"
      >
        <span>
          Enter a city and state
        </span>
      </label>
      <button
        class="submitButton"
      >
        submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
